### PR TITLE
fix result tab

### PR DESCRIPTION
**Pull request by Tianhao-Gu**
Screen Shot 2019-09-04 at 11 14 04 PM

### CODE REVIEW

**Comment by Tianhao-Gu** [comment]
Fix job logs as well. 

Screen Shot 2019-09-04 at 11 38 42 PM

**Comment by Tianhao-Gu** [comment]
Fix job status as well

Screen Shot 2019-09-05 at 12 40 06 AM

**Comment by Tianhao-Gu** [comment]
Fix job status as well

Screen Shot 2019-09-05 at 12 40 06 AM

File: src/biokbase/narrative/jobs/jobmanager.py [hunk 1]
```diff
@@ -874,6 +874,8 @@ def _get_all_job_states(self, job_ids=None):
                 state['run_id'] = self._running_jobs[job_id]['job'].run_id
                 if status == 'finished':
                     self._completed_job_states[state['job_id']] = dict(state)
+                state['job_input'] = literal_eval(state.get('job_input', '{}'))
```
Comment (briehl) [inline src/biokbase/narrative/jobs/jobmanager.py]: This has bothered me for a few days, and I think I finally figured out why - the API shouldn't be returning goofy non-standard JSON strings that require an AST to eval, but the actual unpacked structure. I think that should be revisited.

File: nbextensions/appCell2/widgets/tabs/jobStateViewer.js [hunk 2]
```diff
@@ -26,14 +26,26 @@ define([
...
+                break;
             case 'suspend':
                 label = 'error';
                 color = 'red';
                 break;
+            case 'error':
+                label = 'error';
+                color = 'red';
+                break;
             case 'canceled':
```
Comment (briehl) [inline nbextensions/appCell2/widgets/tabs/jobStateViewer.js]: Is "canceled" still an option from the API? What are the states that are available?

File: nbextensions/appCell2/widgets/appCellWidget.js [hunk 3]
```diff
@@ -1786,9 +1786,9 @@ define([
                     // to inform us about what processing stage the
                     // error occurred in -- we need to inspect the job state.
                     var errorStage;
-                    if (jobState.exec_start_time) {
+                    if (jobState.running) {
```
Comment (briehl) [inline nbextensions/appCell2/widgets/appCellWidget.js]: Can this be figured out by the job status? And not guessing from which fields are present?

File: nbextensions/appCell2/widgets/tabs/jobStateViewer.js (hunk 2 above)
Comment (Tianhao-Gu) [reply nbextensions/appCell2/widgets/tabs/jobStateViewer.js]: no. `canceled` status is no longer available. I updated the state viewer code to remove all `canceled` instances.

File: src/biokbase/narrative/jobs/jobmanager.py (hunk 1 above)
Comment (Tianhao-Gu) [reply src/biokbase/narrative/jobs/jobmanager.py]: 👍 updated ee2 API.

File: nbextensions/appCell2/widgets/appCellWidget.js (hunk 3 above)
Comment (Tianhao-Gu) [reply nbextensions/appCell2/widgets/appCellWidget.js]: hmm it's already in `jobState.status` case block.

**Comment by briehl** [comment]
I think that failing frontend test is a red herring right now. There's other things to do on that branch, anyway, that should ameliorate tests. Let's go ahead and merge this in now.

**Pull request merged**